### DISCUSSION TITLE
Match a Github release with a version prefix and that uses underscores

### DISCUSSION THREAD

**Discussion by SatoshiGuacamole** [renovatebot/renovate#28487] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitHub action v40.1.9

### Please tell us more about your question or problem

I would like to use Renovate to manage my Docker dependency on [Crypto++](https://github.com/weidai11/cryptopp/releases). I've tried github-tags and github-releases as the datasourceTemplate. The version tags of Crypto++ are prefixed with `CRYPTOPP_` and uses underscores instead of periods in the version number, like so: `CRYPTOPP_8_9_0`.

**Dockerfile**

```Dockerfile
ARG CRYPTOPP_VERSION=8.8.0

RUN curl -fsSL "https://github.com/weidai11/cryptopp/archive/refs/tags/CRYPTOPP_${CRYPTOPP_VERSION//./_}/cryptopp${CRYPTOPP_VERSION//./}.tar.gz"
```

**renovate.json5**

```json5
{
    baseBranches: [
        "main",
    ],
    enabledManagers: [
        "custom.regex",
    ],
    customManagers: [
        {
            depNameTemplate: "Crypto++",
            fileMatch: ["^Dockerfile$"],
            matchStrings: ["ARG CRYPTOPP_VERSION=(?<currentValue>.*?)\\n"],
            datasourceTemplate: "github-releases",
            packageNameTemplate: "weidai11/cryptopp",
        },
    ],
    packageRules: [
        {
            matchPackageNames: ["weidai11/cryptopp"],
            versionCompatibility: "^(?<compatibility>CRYPTOPP_)(?<version>.*)$",
            versioning: "regex:^(?<major>\\d+)_(?<minor>\\d+)_(?<patch>\\d+)$",
        },
    ],
}
```

The above configuration does not create a PR if I want to upgrade from `CRYPTOPP_8_8_0`.

[A related topic that I found but that does not have the underscore requirement](https://github.com/renovatebot/renovate/issues/7297).

**Comment by rarkins** [comment]
We might be able to figure out a way.

First of all, is there anything stopping you from doing one of these?

```
ARG CRYPTOPP_VERSION=8_8_0
ARG CRYPTOPP_VERSION=CRYPTO_PP8_8_0
```

If you took this approach then you could define regex versioning which applies the same way to both your repo strings as well as the github release strings.

**Reply by nedix** [reply, marked as answer]
That put me on the right track.

**Dockerfile**

```Dockerfile
ARG CRYPTOPP_VERSION=8_8_0

RUN curl -fsSL "https://github.com/weidai11/cryptopp/archive/refs/tags/CRYPTOPP_${CRYPTOPP_VERSION}/cryptopp${CRYPTOPP_VERSION//_/}.tar.gz"
```

**renovate.json5**

```json5
{
    baseBranches: [
        "main",
    ],
    enabledManagers: [
        "custom.regex",
    ],
    customManagers: [
        {
            depNameTemplate: "Crypto++",
            fileMatch: ["^Dockerfile$"],
            matchStrings: ["ARG CRYPTOPP_VERSION=(?<currentValue>.*?)\\n"],
            datasourceTemplate: "github-releases",
            packageNameTemplate: "weidai11/cryptopp",
        },
    ],
    packageRules: [
        {
            description: "Custom versioning for Crypto++",
            matchPackageNames: ["weidai11/cryptopp"],
            extractVersion: "^CRYPTOPP_(?<version>.*)$",
            versioning: "regex:^(?<major>\\d+)_(?<minor>\\d+)_(?<patch>\\d+)$",
        },
    ],
}
```

Thank you for your help @rarkins
